framework/src/modules/http_api/controllers/node.js

```diff
@@ -15,13 +15,65 @@
 'use strict';
 
 const _ = require('lodash');
+const { getAddressFromPublicKey } = require('@liskhq/lisk-cryptography');
 const checkIpInList = require('../helpers/check_ip_in_list');
 const apiCodes = require('../api_codes');
 const swaggerHelper = require('../helpers/swagger');
 
 let library;
 let epochTime;
-let fees;
+
+function _filterTransactions(transactions, filters) {
+	const { limit, offset, sort } = filters;
+	let filteredTransactions = transactions;
+	if (filters.id) {
+		filteredTransactions = filteredTransactions.filter(
+			transaction => transaction.id === filters.id,
+		);
+	}
+	if (filters.senderId || filters.senderPublicKey) {
+		filteredTransactions = filteredTransactions.filter(transaction => {
+			if (filters.senderId) {
+				return (
+					getAddressFromPublicKey(transaction.senderPublicKey) ===
+					filters.senderId
+				);
+			}
+			return transaction.senderPublicKey === filters.senderPublicKey;
+		});
+	}
+	if (filters.type !== null && filters.type !== undefined) {
+		filteredTransactions = filteredTransactions.filter(
+			transaction => transaction.type === filters.type,
+		);
+	}
+	if (sort) {
```

Comment:
I remember we discussed sorting for `nonce` and by default, it will be `asc` sorted.  are we not adding the `nonce` for sorting?

Reply:
I think we decided not to include the nonce because if only nonce is provided, it does not provide useful information.